user-manager/src/test/java/com/qpay/usermanager/controllers/MerchantControllerIntergrationTest.java

```diff
@@ -0,0 +1,168 @@
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+
+@WebMvcTest(controllers = MerchantController.class)
+class MerchantControllerIntergrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private MerchantServiceImpl merchantService;
+
+    private final MerchantEntity MERCHANT_ENTITY = MerchantEntity.builder()
+            .name("bob")
+            .email("bob@gmail.com")
+            .password("1234")
+            .walletId(1L)
+            .build();
+
+    private final MerchantCreation MERCHANT_CREATION = new MerchantCreation("bob", "bob@gmail.com", "word", 1L);
+
+    @Test
+    void should_returnMerchant() throws Exception {
+        //given
+        given(merchantService.getMerchantById(1L)).willReturn(MERCHANT_ENTITY);
+
+        //when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders.get(PathsUtils.MERCHANTS_PATH + "/{id}", 1L))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        //then
+        var expectedResponseBody = objectMapper.writeValueAsString(MERCHANT_ENTITY);
+        assertThat(responseBody).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    void should_returnException_when_merchantIdNotFound() throws Exception {
+        //given
+        given(merchantService.getMerchantById(1L)).willThrow(new NoMerchantFoundException("not found"));
+
+        //when
+        var status = mockMvc
+                .perform(MockMvcRequestBuilders.get(PathsUtils.MERCHANTS_PATH + "/{id}", 1L))
+                .andReturn()
+                .getResponse()
+                .getStatus();
+
+        //then
+        assertThat(status).isEqualTo(HttpStatus.NOT_FOUND.value());
+    }
+
+    @Test
+    void should_returnMerchant_when_merchantAdded() throws Exception {
+        //given
+        given(merchantService.addMerchant(MERCHANT_CREATION)).willReturn(MERCHANT_ENTITY);
+
+        //when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders
+                        .post(PathsUtils.MERCHANTS_PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(MERCHANT_CREATION)))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        //then
+        var expectedResponseBody = objectMapper.writeValueAsString(MERCHANT_ENTITY);
+        assertThat(responseBody).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    void should_returnException_when_addMerchantInputInvalid() throws Exception {
+        //given
+        var invalidMerchantCreation = new MerchantCreation("", "", "", 1L);
```

Comment:
use builder. Otherwise it's hard to read the code. It is not clear what these fields are.